app/src/main/java/com/marverenic/music/player/PlayerService.java

```diff
@@ -497,6 +500,26 @@ public void setQueue(List<Song> newQueue, int newPosition) throws RemoteExceptio
         }
 
         @Override
+        public void beginBigQueue() throws RemoteException {
+            mQueueTemp = new ArrayList<>();
+        }
+
+        @Override
+        public void sendQueueChunk(List<Song> chunk) throws RemoteException {
+            Preconditions.checkNotNull(mQueueTemp).addAll(chunk);
+        }
+
+        @Override
+        public void endBigQueue(int flag, int newPosition) throws RemoteException {
+            List<Song> queue = Preconditions.checkNotNull(mQueueTemp);
+            mQueueTemp = null;
+            if (flag == PlayerController.FLAG_SET_QUEUE)
+                setQueue(queue, newPosition);
+            else if (flag == PlayerController.FLAG_EDIT_QUEUE)
+                editQueue(queue, newPosition);
```

Comment:
I think this should take in a boolean (i.e. `editQueue`) instead of the flag parameter. Also, I'd prefer to use curly braces around the bodies here, just for consistency with the rest of Jockey's code.